### PR TITLE
Comprehensive Refactoring and Optimization of Project for Enhanced Readability, Efficiency, and Robustness

### PR DESCRIPTION
**Pull request by RoachxD**
This Pull Request encompasses a series of improvements and refinements to the project's structure, functionality, and code quality, with the objective of enhancing readability, efficiency, and error handling.

The primary contributions include:

- Optimization of `setup.py`: Refactoring the file for increased efficiency and readability, including utilization of the `Path` object from `pathlib` for better file path handling, simplification of `RegEx` patterns, and employment of `f-string` for string formatting.
- Refactoring of extension scripts: Adhering to Python's naming conventions and optimizing various aspects of the code, such as list comprehensions and `f-string` usage, to improve consistency, readability, and efficiency.
- Improved formatting of `conf.py`: Enhancing the readability and consistency of the file by removing blank lines, refining formatting, and employing list indexing with `re.search`.
- Implementation of the `all` variable: Explicitly defining public members for import, promoting a clean API and preventing unintended imports of private or internal members.
- Refinement of `request.py`: Removing unnecessary imports, utilizing the params keyword argument for improved readability, and streamlining error handling through `response.raise_for_status()` and combined exception handling.
- Division of `ESteamCurrency` into two classes: Separating supported and unsupported currencies to facilitate better error handling and functionality, while acknowledging potential limitations in regards to IDEs autocompletion.
- Introduction of a new exception: Adding `SteamUnsupportedCurrency` exception to handle cases where an unsupported currency is encountered.
- Implementation of float parsing for all currencies: Incorporating the `re` package for accurate parsing of both supported and unsupported currencies, and optimizing various aspects of the code, such as type hints, list comprehensions, and conditional statements, for improved readability and efficiency.

These cumulative enhancements result in a more streamlined, robust, and maintainable codebase that adheres to best practices and provides a solid foundation for future development.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,13 +23,13 @@
...
+    download_url=f"https://github.com/offish/steam_community_market/tarball/v{version}",
     packages=["steam_community_market"],
     classifiers=[
         "Programming Language :: Python :: 3",
         "License :: OSI Approved :: MIT License",
         "Operating System :: OS Independent",
     ],
     install_requires=["requests"],
-    python_requires='>=3.6',
+    python_requires=">=3.6",
```
Comment (Gobot1234) [inline setup.py]: This is no longer correct due to the usage of the new union syntax

File: steam_community_market/market.py [hunk 2]
```diff
@@ -46,222 +58,219 @@ def get_overview(self, name: str, app_id) -> dict:
...
-        :return: The volume if success, :class:`None` otherwise.
-        :rtype: Optional[:class:`int`]
+        :type app_id: :class:`int`, list[int | AppID], :class:`AppID`
+        :param support_lists: Whether or not to support lists.
+        :type support_lists: :class:`bool`
+        :return: The value of the AppID.
+        :rtype: :class:`int`
 
-        .. versionadded:: 1.2.0
+        .. versionadded: 1.2.4
```
Comment (Gobot1234) [inline steam_community_market/market.py]: This semi-colon shouldn't be removed

File: setup.py (hunk 1 above)
Comment (RoachxD) [reply setup.py]: Correct, modified to `>=3.10`.

File: steam_community_market/market.py (hunk 2 above)
Comment (RoachxD) [reply steam_community_market/market.py]: Fixed, many thanks for pointing it out!

**Pull request merged**